src/main/java/com/epam/processor/DataProcessor.java

```diff
@@ -1,123 +1,240 @@
-     */
-    public Map<String, Long> getCountByRoadSurfaceCondition(){
-        return null;
-    }
-
-    /**
-     * To match streaming operations result, return type is a java collection instead of multimap
-     * @return
-     */
-    public Map<String, List<String>> getAccidentIdsGroupedByAuthority(){
-        return null;
-    }
-
-}
+package com.epam.processor;
+
+import com.epam.data.AccidentsDataLoader;
+import com.epam.data.RoadAccident;
+import com.epam.data.RoadAccidentBuilder;
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.Multimap;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * This is to be completed by mentees
+ */
+public class DataProcessor {
+	private static final String ACCIDENTS_CSV = "src/main/resources/DfTRoadSafety_Accidents_2009.csv";
+
+	private final List<RoadAccident> roadAccidentList;
+
+	public DataProcessor(List<RoadAccident> roadAccidentList) {
+		this.roadAccidentList = roadAccidentList;
+	}
+
+	// First try to solve task using java 7 style for processing collections
+
+	/**
+	 * Return road accident with matching index
+	 * 
+	 * @param index
+	 * @return
+	 */
+	public RoadAccident getAccidentByIndex7(String index) {
+		for (RoadAccident roadAccident : roadAccidentList) {
+			if (roadAccident.getAccidentId().equals(index)) {
+				return roadAccident;
+			}
+		}
+		return null;
+	}
+
+	/**
+	 * filter list by longtitude and latitude values, including boundaries
+	 * 
+	 * @param minLongitude
+	 * @param maxLongitude
+	 * @param minLatitude
+	 * @param maxLatitude
+	 * @return
+	 */
+	public Collection<RoadAccident> getAccidentsByLocation7(float minLongitude,
+			float maxLongitude, float minLatitude, float maxLatitude) {
+		List<RoadAccident> list = new ArrayList<>();
+		for (RoadAccident roadAccident : roadAccidentList) {
+			if (roadAccident.getLongitude() >= minLongitude
+					&& roadAccident.getLongitude() <= maxLongitude
+					&& roadAccident.getLatitude() >= minLatitude
+					&& roadAccident.getLatitude() <= maxLatitude) {
+				list.add(roadAccident);
+			}
+		}
+		return list;
+	}
+
+	/**
+	 * count incidents by road surface conditions ex: wet -> 2 dry -> 5
+	 * 
+	 * @return
+	 */
+	public Map<String, Long> getCountByRoadSurfaceCondition7() {
+		Map<String, Long> map = new HashMap<>();
+		long dry = 0, wet = 0;
+
+		for (RoadAccident roadAccident : roadAccidentList) {
+			if (roadAccident.getRoadSurfaceConditions().equals("2")
+					|| roadAccident.getRoadSurfaceConditions()
+							.equalsIgnoreCase("wet")) {
+				wet++;
+			} else if (roadAccident.getRoadSurfaceConditions().equals("5")
+					|| roadAccident.getRoadSurfaceConditions()
+							.equalsIgnoreCase("dry")) {
+				dry++;
+			}
+		}
+		map.put("Dry", dry);
+		map.put("Wet", wet);
+		return map;
+	}

```

Comment:
@zero1608 
2 & 5 are count for dry & wet. They are not conditions. And that is just example.
Please try to implement it in a generic way so any type of condition can come in output.